Add docs for PSAT for Kubelet Image Credential Providers alpha

aramase opened the pull request:
Doc update for KEP 4412: https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/4412-projected-service-account-tokens-for-kubelet-image-credential-providers/README.md
KEP issue: https://github.com/kubernetes/enhancements/issues/4412

rayandas commented:
Hello @aramase  :wave: please take a look at Documenting for a release - PR Ready for Review to get your PR ready for review before Tuesday 25th March 2025 18:00 PDT. Thank you!

liggitt commented on content/en/docs/tasks/administer-cluster/kubelet-credential-provider.md:
```diff
@@ -26,11 +26,24 @@ You may be interested in using this capability if any of the below are true:
+## Service Account Token for Image Pulls
+{{< feature-state feature_gate_name="KubeletServiceAccountTokenForCredentialProviders" >}}
+
+Starting from Kubernetes v1.33, the kubelet can be configured to send the projected service account token bound to the pod for which the image pull is being performed to the credential provider plugin. This allows the plugin to exchange the token for credentials to access the image registry. 
```
the token isn't the projected token, it is distinct to the credential provider

```suggestion
Starting from Kubernetes v1.33, the kubelet can be configured to send a service account token bound to the pod for which the image pull is being performed to the credential provider plugin. This allows the plugin to exchange the token for credentials to access the image registry. 
```

liggitt commented on content/en/docs/tasks/administer-cluster/kubelet-credential-provider.md:
```diff
@@ -26,11 +26,24 @@ You may be interested in using this capability if any of the below are true:
+
+To enable this feature, the `KubeletServiceAccountTokenForCredentialProviders` feature gate must be enabled on the kubelet and the `tokenAttributes` field must be set in the `CredentialProviderConfig` file for the plugin. The `tokenAttributes` field contains information about the service account token that will be passed to the plugin, including the intended audience for the token and whether the ...
+
+You may be interested in using this feature if any of the below are true:
```
```suggestion
Using service account token credentials can enable the following use-cases:
```

liggitt commented on content/en/docs/tasks/administer-cluster/kubelet-credential-provider.md:
```diff
@@ -26,11 +26,24 @@ You may be interested in using this capability if any of the below are true:
+
+You may be interested in using this feature if any of the below are true:
+
+* Avoid needing a kubelet/node based identity to pull images from a registry.
```
```suggestion
* Avoid needing a kubelet/node-based identity to pull images from a registry.
```

liggitt commented on content/en/docs/tasks/administer-cluster/kubelet-credential-provider.md:
```diff
@@ -26,11 +26,24 @@ You may be interested in using this capability if any of the below are true:
+You may be interested in using this feature if any of the below are true:
+
+* Avoid needing a kubelet/node based identity to pull images from a registry.
+* Allow workloads to pull images based on their own runtime identity without long lived/persisted secrets.
```
```suggestion
* Allow workloads to pull images based on their own runtime identity without long-lived/persisted secrets.
```

liggitt commented:
looks good overall, just a few suggestions

liggitt commented:
lgtm for sig-auth

tengqm commented on content/en/docs/tasks/administer-cluster/kubelet-credential-provider.md:
```diff
@@ -26,11 +26,24 @@ You may be interested in using this capability if any of the below are true:
+
+Starting from Kubernetes v1.33, the kubelet can be configured to send a service account token bound to the pod for which the image pull is being performed to the credential provider plugin. This allows the plugin to exchange the token for credentials to access the image registry.
+
+To enable this feature, the `KubeletServiceAccountTokenForCredentialProviders` feature gate must be enabled on the kubelet and the `tokenAttributes` field must be set in the `CredentialProviderConfig` file for the plugin. The `tokenAttributes` field contains information about the service account token that will be passed to the plugin, including the intended audience for the token and whether the ...
```
Please help wrap these long "lines" properly as we do for other paragraphs.
https://sembr.org/

aramase replied:
pushed an update, LMK what you think!

aramase commented:
/assign @tengqm

aramase commented:
> Hello @aramase 👋! I'm reaching out from the Docs team. Just checking in as we approach **Docs Freeze** on **8th April, 2025 18:00 PDT**. This documentation appears to still be under review. To meet the Docs Freeze, this PR must have a technical review as well as `lgtm` and `approve` labels applied, without any unaddressed comments or concerns from SIG Docs. The status of this enhancement is marked as `At risk for docs freeze`. Thank you!

@rayandas I have a `lgtm` from sig-auth changes (https://github.com/kubernetes/website/pull/49884#issuecomment-2755047900), waiting for `lgtm` + `approve` from the docs team.

tengqm commented:
Upstream code changes seem complete. Technical LGTM.
/lgtm
/approve
